### PR TITLE
Land enhancement

### PR DESCRIPTION
**Pull request by dakejahl**
This PR fixes two problems with _Land_. See #9691 & #9768

**Problems**
1. _Land_ can cause vehicle to flip upon land if the command was issued while the vehicle was moving quickly. 

2. _Land_ descent velocity was only **MPC_LAND_SPEED**. 

**Solutions**
1. Vehicle will return to LAT/LON location where the _Land_ command was issued and begin descending from there. In the future, I want to implement a **_BRAKE_** setpoint type, but this is a good fix until that work is done.

2. The vehicle now respects MPC_LAND_ALT1 and MPC_LAND_ALT2 to control descent velocities.

I have simulation tested and physically flight tested this on our quadcopter and have seen no problems. I would love if someone could please test this with a fixed wing aircraft to verify. 

Please let me know if you have any questions, thanks!
Jake

### CODE REVIEW

File: src/modules/mc_pos_control/mc_pos_control_main.cpp [hunk 1]
```diff
@@ -1450,8 +1450,20 @@ MulticopterPositionControl::control_non_manual()
 	/* use constant descend rate when landing, ignore altitude setpoint */
 	if (_pos_sp_triplet.current.valid
 	    && _pos_sp_triplet.current.type == position_setpoint_s::SETPOINT_TYPE_LAND) {
-		_vel_sp(2) = _land_speed.get();
 		_run_alt_control = false;
+
```
Comment (MaEtUgR) [inline src/modules/mc_pos_control/mc_pos_control_main.cpp]: This whole block is duplicating the existing functionality: https://github.com/PX4/Firmware/blob/master/src/modules/mc_pos_control/mc_pos_control_main.cpp#L2439-L2443
Did it not work correctly for you?

Comment (dakejahl) [reply src/modules/mc_pos_control/mc_pos_control_main.cpp]: Look at the one liner that I removed.

 _vel_sp(2) = _land_speed.get();

Clearly this is overriding the current Z vel setpoint when the setpoint type is LAND.

Also, always limiting veritial velocity based upon height does not seem right. Those parameters are specified with the word 'LAND' and thus should only affect flight characteristics while in the LAND mode (this is my opinion).

**Comment by dakejahl** [comment]
@Stifael Behavior in **_mc_pos_control_main.cpp_** remains the same if there is no local position.
```
		if (abs(_local_pos.z) > _slow_land_alt1.get()) {
			_vel_sp(2) = _vel_max_down.get();

		} else if (abs(_local_pos.z) > _slow_land_alt2.get()) {
			float velocity_scaling = (abs(_local_pos.z) - _slow_land_alt2.get()) / (_slow_land_alt1.get() - _slow_land_alt2.get());
			_vel_sp(2) = _land_speed.get() + velocity_scaling * (_vel_max_down.get() - _land_speed.get());

		} else {
			_vel_sp(2) = _land_speed.get();
		}
```

Does a failsafe event override navigator modes?

**Comment by dakejahl** [comment]
Actually, the way the code is _currently_ it seems like if local position was lost it would never make it into the first `if` statement for `land`

```
	if (_pos_sp_triplet.current.valid
	    && _pos_sp_triplet.current.type == position_setpoint_s::SETPOINT_TYPE_LAND) {
```

**Comment by dakejahl** [comment]
@dagar

**Comment by Stifael** [comment]
@dakejahl with your changes, the first stages are not of type `SETPOINT_TYPE_LAND`, they are of type `SETPOINT_TYPE_POSITION` (https://github.com/PX4/Firmware/pull/9772/files#diff-7a824e3d1e9a5420071c8c9bbe27b8e7R643). 
Have you tried in SITL to do fast forward flight and then turn of GPS?

**Comment by Stifael** [comment]
a break would also be helpful for this "feature" request: https://github.com/PX4/Firmware/issues/9741#issuecomment-399912835

**Comment by dakejahl** [comment]
Please see #9804

**Pull request closed without merging**